### PR TITLE
Checkbox criteria

### PR DESCRIPTION
**Pull request by WChrisK**

### CODE REVIEW

File: app/assets/stylesheets/checkbox_criteria.scss [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    margin-top: 1em;
+  }
+
+  #checkbox_criteria_preview {
+    border-top: 1px solid $grey;
+    margin-top: 1.5em;
+    padding-top: 1.5em;
+  }
+
+  /* TODO - Is this needed? Why did flexible_criteria.scss have it? */
```
Comment (WChrisK) [inline app/assets/stylesheets/checkbox_criteria.scss]: Noting this that I'll be checking to see if this needs to be axed or not.

File: app/models/checkbox_criterion.rb [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+class CheckboxCriterion < Criterion
+  self.table_name = 'checkbox_criteria'
+
+  belongs_to :assignment, counter_cache: true
+  has_many :criterion_ta_associations, as: :criterion, dependent: :destroy
+  has_many :marks, as: :markable, dependent: :destroy
+  has_many :tas, through: :criterion_ta_associations
+  has_many :test_scripts, as: :criterion
+
+  # TODO - Remove I18n if it allows it
```
Comment (WChrisK) [inline app/models/checkbox_criterion.rb]: _Note to self_: Remove this

File: app/models/assignment.rb [hunk 3]
```diff
@@ -682,6 +688,8 @@ def criterion_class
       FlexibleCriterion
     elsif marking_scheme_type == MARKING_SCHEME_TYPE[:rubric]
       RubricCriterion
+    elsif marking_scheme_type == MARKING_SCHEME_TYPE[:checkbox]
```
Comment (david-yz-liu) [inline app/models/assignment.rb]: remove

File: app/models/checkbox_criterion.rb [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+  def set_mark_by_criterion(mark_to_change, mark_value)
+    if mark_value == 'nil'
+      mark_to_change.mark = nil
+    else
+      mark_to_change.mark = mark_value.to_f
+    end
+    mark_to_change.save
+  end
+
+  # Instantiate a FlexibleCriterion from a CSV row and attach it to the supplied
```
Comment (david-yz-liu) [inline app/models/checkbox_criterion.rb]: comment

File: config/locales/fr.yml [hunk 5]
```diff
@@ -1119,6 +1120,34 @@ fr:
       peer_visible: "Rendre visible à leurs auteurs pairs"
       visibility_error: "Sélectionnez au moins une case dans la section de visibilité"
 
+    checkbox_criteria:
+      title: "Critères d' Checkbox"
+      move_up: "Monter"
```
Comment (GitGeminiano) [inline config/locales/fr.yml]: Most of these strings are shared among criteria. Perhaps, we can put them under criteria which is immediately above.

Comment (WChrisK) [inline config/locales/fr.yml]: I was thinking the same thing as well, there's probably a chunk of them that could go under the criteria. If I recall, there was only 6 that differed between them.

**Pull request merged**